core/src/main/scala/akka/persistence/r2dbc/journal/JournalDao.scala

```diff
@@ -102,9 +103,9 @@ private[r2dbc] class JournalDao(journalSettings: R2dbcSettings, connectionFactor
 
     val insertEventWithTransactionTimestampSql = {
       if (journalSettings.dbTimestampMonotonicIncreasing)
-        sql"$baseSql transaction_timestamp())"
+        sql"$baseSql transaction_timestamp()) RETURNING transaction_timestamp() AS db_timestamp"
       else
-        sql"$baseSql GREATEST(transaction_timestamp(), $timestampSubSelect))"
+        sql"$baseSql GREATEST(transaction_timestamp(), $timestampSubSelect)) RETURNING transaction_timestamp() AS db_timestamp"
```

Comment:
Doesn't this mean it could insert the subselect one but return the transaction timestamp -> mismatch?

Reply:
`RETURNING db_timestamp` should work

Reply:
thanks, changed to that